subsys/bluetooth/audio/cap_stream.c

```diff
@@ -15,6 +21,12 @@ static void cap_stream_configured_cb(struct bt_bap_stream *bap_stream,
 							bap_stream);
 	struct bt_bap_stream_ops *ops = cap_stream->ops;
 
+	LOG_DBG("%p", cap_stream);
+
+	if (IS_ENABLED(CONFIG_BT_CAP_INITIATOR)) {
+		bt_cap_initiator_codec_configured(cap_stream);
+	}
+
 	if (ops != NULL && ops->configured != NULL) {
 		ops->configured(bap_stream, pref);
 	}
```

Comment:
It looks like dependency is mixed here. CAP initiator code calls `bt_cap_stream_ops_register_bap` which registers this callback and from this callback, the `bt_cap_initiator_codec_configured` is called.
Probably, the cap_initiator.c shall have it's own bt_bap_stream_ops.

Reply:
> Probably, the cap_initiator.c shall have it's own bt_bap_stream_ops.

Yes, this code is from before the BAP rename, where the callback structure was called `bt_audio_stream_ops` and was more generic. 

CAP is effectively on top of BAP, so the BAP stream ops are registered by CAP, and the CAP stream parameters are registered by the user of CAP. 

So CAP registers a set of callbacks for the BAP stream, and uses those callbacks to execute the CAP procedures, while also calling the CAP stream callbacks registered by the user of CAP. 

I am not sure I understand the issue you may be on to here, so please elaborate a bit more :)

Reply:
Ah, I thought it's on BAP level (`bap_stream.c`), meaning `bt_cap_initiator_codec_configured` is called from BAP level. If it's `cap_stream.c` it's OK :D